### ISSUE TITLE
unrecognized selector sent to instance error

### ISSUE THREAD

**Issue by slcott** [evermeer/EVReflection#84]
I am trying to deserialize an xml document to a deeply nested class hierarchy. Your library is working for very simple cases but as I test with deeper nesting it causes an error when I try to use the deserialized object:

```
[_TtGCs29_NativeDictionaryStorageOwnerSSGVs10DictionarySSCSo8NSObject__ objectAtIndex:]: unrecognized selector sent to instance 0x7fb301625240
```

It looks like the bug could be that it is trying to set an object using a dictionary:

```
            // Call your own object validators that comply to the format: validate<Key>:Error:
            do {
               var setValue: AnyObject? = value
                try anyObject.validateValue(&setValue, forKey: key)
                anyObject.setValue(setValue, forKey: key)
            } catch _ {
```

Debugger shows this:

```
(lldb) po key
"directionType"

(lldb) po typeInObject
▿ Optional<String>
  - Some : "Array<DirectionType>"

(lldb) po setValue
▿ Optional<AnyObject>
  ▿ Some : 1 elements
    ▿ [0] : 2 elements
      - .0 : words
      ▿ .1 : 4 elements
        ▿ [0] : 2 elements
          - .0 : __text
          - .1 : Ziemlich und mit Ausdruck
        ▿ [1] : 2 elements
          - .0 : _fontWeight
          - .1 : bold
        ▿ [2] : 2 elements
          - .0 : _defaultY
        ▿ [3] : 2 elements
          - .0 : _fontSize
```

Is this possibly a bug or a mistake in my use of the library?

I have attached the test case.

[Archive.zip](https://github.com/evermeer/EVReflection/files/275921/Archive.zip)

**Comment by evermeer** [comment]
Complexity should not be an issue. 
There are 2 mistakes in your code. 
- The field _fontWeight is defined as an Int but it's a String. But then that will only let it stay 0
- The words poperty is an array of objects but in your data declaration it's just an object. If you add an extra [] around the word properties it will succeed.
  But still... It should not crash like this. I thought my code would even allow this structure. I did write some code for if you try to put an object in an array it would first put the object in an array and then assign it to the array. I will do some investigation what has happened here.

**Comment by slcott** [comment]
The second bullet point is the important problem for me precisely. I use XMLDictionary first to convert xml to a NSDictionary. Xml (without the schema) does not show arrays like json. If EVReflection can dynamically determine either is an object or an array that would be best for me. 

I tried coding it myself, but need to study the code more. Anyway, this is where I started to attack the problem inside: `public static func setObjectValue<T where T: NSObject>(anyO`

```
        if typeInObject == "Struct" {
            anyObject.setValue(value, forUndefinedKey: key)
        }
        else if (typeInObject?.hasPrefix("Array<"))! {
            let setValue: AnyObject? = value
            if let setValueDictionary = (setValue as? NSDictionary) {
                let newArray = T() // T is of type Array<EVObject>

                var genericTypeString = typeInObject!.substringFromIndex((typeInObject!.componentsSeparatedByString("<") [0] + "<").endIndex)
                genericTypeString = genericTypeString.substringToIndex(genericTypeString.endIndex.predecessor())
                let newObject = fromDictionary(setValueDictionary, anyobjectTypeString: genericTypeString)

                EVReflection.setPropertiesfromDictionary(setValueDictionary, anyObject: newArray, conversionOptions: .DefaultDeserialize)
                if let array = newArray as? NSArray {

                    array.performSelector(#selector(NSArray.arrayByAddingObject(_:)), withObject: newObject)
                }
                anyObject.setValue(newArray, forKey: key)
            }
        }
        else {
            if !valid {
                (anyObject as? EVObject)?.addStatusMessage(.InvalidType, message: "\(anyObject.dynamicType) `\(key)` type, `\(type), doesn't match expected type.")
                print("WARNING: \(anyObject.dynamicType) `\(key)` type, `\(type), doesn't match expected type.")
                anyObject.setValue(theValue, forUndefinedKey: key)
                return
            }
```

**Comment by evermeer** [comment]
I just released version 2.32.2 which will solve your issue. The problem was that there where more than 1 level of object to array fix at the same time. directionType was an object that needed to be an array and words also was an object that needed to be an array. EVReflection will now support multiple levels of this sort of automatic conversions.
